tests/test_generics.py

```diff
@@ -519,6 +520,37 @@ class M(BaseModel):
     del generics
 
 
+def test_generics_reused() -> None:
+    """https://github.com/pydantic/pydantic/issues/11747
+
+    To fix an issue with recursive generics, we introduced a change in 2.11 that would
+    skip caching the parameterized model under specific circumstances. The following setup
+    is an example of where this happens. As a result, we end up with two different `A[float]`
+    classes, although they are the same in practice.
+    When serializing, we check that the value instances are matching the type, but we end up
+    with warnings as `isinstance(value, A[float])` fails.
```

Comment:
`int` not `float`? Also I think the tense might be past tense...

```suggestion
    is an example of where this would happen. As a result, we ended up with two different `A[int]`
    classes, although they were the same in practice.
    When serializing, we check that the value instances are matching the type, but we ended up
    with warnings as `isinstance(value, A[int])` fails.
```